internal/sirius/my_details_test.go

```diff
@@ -123,3 +133,131 @@ func TestMyDetails(t *testing.T) {
 		})
 	}
 }
+
+func TestEditMyDetails(t *testing.T) {
+	pact := &dsl.Pact{
+		Consumer:          "sirius-user-management",
+		Provider:          "sirius",
+		Host:              "localhost",
+		PactFileWriteMode: "merge",
+		LogDir:            "../../logs",
+		PactDir:           "../../pacts",
+	}
+	defer pact.Teardown()
+
+	testCases := map[string]struct {
+		phoneNumber              string
+		setup                    func()
+		cookies                  []*http.Cookie
+		expectedValidationErrors ValidationErrors
+		expectedError            error
+	}{
+		"OK": {
+			phoneNumber: "01210930320",
+			setup: func() {
+				pact.
+					AddInteraction().
+					Given("User exists").
+					UponReceiving("A request to change my phone number").
+					WithRequest(dsl.Request{
+						Method: http.MethodPut,
+						Path:   dsl.String("/api/v1/users/47/updateTelephoneNumber"),
+						Headers: dsl.MapMatcher{
+							"Content-type":        dsl.String("application/json"),
+							"X-XSRF-TOKEN":        dsl.String("abcde"),
+							"Cookie":              dsl.String("XSRF-TOKEN=abcde; Other=other"),
+							"OPG-Bypass-Membrane": dsl.String("1"),
+						},
+						Body: map[string]string{
+							"phoneNumber": "01210930320",
+						},
+					}).
+					WillRespondWith(dsl.Response{
+						Status:  http.StatusOK,
+						Headers: dsl.MapMatcher{"Content-Type": dsl.String("application/json")},
+						Body:    dsl.Match(myDetailsResponse{}),
+					})
+			},
+			cookies: []*http.Cookie{
+				{Name: "XSRF-TOKEN", Value: "abcde"},
+				{Name: "Other", Value: "other"},
+			},
+			expectedValidationErrors: nil,
+		},
+
+		"BadRequest": {
+			phoneNumber: "invalid phone number",
+			setup: func() {
+				pact.
+					AddInteraction().
+					Given("User exists").
+					UponReceiving("An invalid request to change my phone number").
+					WithRequest(dsl.Request{
+						Method: http.MethodPut,
+						Path:   dsl.String("/api/v1/users/47/updateTelephoneNumber"),
+						Headers: dsl.MapMatcher{
+							"X-XSRF-TOKEN":        dsl.String("abcde"),
+							"Cookie":              dsl.String("XSRF-TOKEN=abcde; Other=other"),
+							"OPG-Bypass-Membrane": dsl.String("1"),
+						},
+						Body: map[string]string{
+							"phoneNumber": "invalid phone number",
+						},
+					}).
+					WillRespondWith(dsl.Response{
+						Status:  http.StatusBadRequest,
+						Headers: dsl.MapMatcher{"Content-Type": dsl.String("application/json")},
```

Comment:
Does Sirius actually return this header, I know for the change password one it doesn't

Reply:
You're quite right, it's `application/problem+json`.